### PR TITLE
fix(list-item): improve drag handle spacing

### PR DESCRIPTION
**Pull request by eriklharper**
**Related Issue:** #11125 

## Summary

This removes extra `inline-start` padding on the list-item's drag handle for all scales and adds an appropriate amount of `inline-end` padding to the handle in `selection-mode="none"` and `selection-appearance="border"` variants.

BEGIN_COMMIT_OVERRIDE
omitted from changelog
END_COMMIT_OVERRIDE

### CODE REVIEW

File: packages/calcite-components/src/components/list-item/list-item.scss [hunk 1]
```diff
@@ -147,6 +147,12 @@
   color: var(--calcite-list-icon-color, var(--calcite-color-text-3));
 }
 
+:host([display-mode="flat"][drag-handle][selection-mode="none"]),
```
Comment (jcfranco) [inline packages/calcite-components/src/components/list-item/list-item.scss:150]: The [`is`](https://developer.mozilla.org/en-US/docs/Web/CSS/:is) function can help simplify this and similar rules:

```scss
:host([display-mode="flat"][drag-handle]:is([selection-mode="none"], [selection-appearance="border"])) {
  .drag-container {
    padding-inline-end: var(--calcite-spacing-xxs);
  }
}
```

File: packages/calcite-components/src/components/list-item/list-item.scss [hunk 2]
```diff
@@ -228,7 +234,7 @@
 }
 :host([scale="m"][display-mode="flat"][drag-handle]) {
   .container {
-    padding-inline-start: var(--calcite-spacing-xxs);
+    padding-inline-start: 0;
```
Comment (jcfranco) [inline packages/calcite-components/src/components/list-item/list-item.scss:237]: Can we simplify by using internal CSS variables?

**Review by jcfranco — APPROVED** [review]
Aside from some suggestions to simplify, this LGTM!

**Note**: clean up can be handled in a follow-up PR

**Comment by jcfranco** [comment]
@ashetland can you review the screenshot diffs?

**Review by ashetland — APPROVED** [review]
🚀🚀🚀

**Pull request merged**